### ISSUE TITLE
--await for healthcheck

### ISSUE THREAD

**Issue by JamesTheAwesomeDude** [docker/compose#8003]
When running `docker-compose up -d`, I'd like to be able to block until the container is actually running and "healthy", if possible.

e.g., a service will wait for its dependencies to be healthy before starting, but I'm kicked back out to the shell (or shell script, as it were,) before waiting for the actual _invocation_ to mature to health.

I propose syntax along the lines of

```sh
#!/bin/sh
docker-compose up -d --await=web &&\
  echo "The service is running and healthy." ||\
  echo "There was a problem bringing up the service."
```

**Comment by JamesTheAwesomeDude** [comment]
While there are [3rd-party hacks](https://github.com/vishnubob/wait-for-it), as well as e.g. simply `until wget --spider localhost:4000 ; do sleep 1 ; done`, it seems a bit strange that there's no _built-in_ way to `await` the healthchecks that we've _already_ so carefully coded into our `docker-compose.yml` files.

**Comment by branchvincent** [comment]
I was also expecting this functionality, imagining a `docker-compose up -d --wait` (similar to [helm](https://helm.sh/docs/intro/using_helm/#helpful-options-for-installupgraderollback)). Our use case is a CI/CD environment where we want to ensure our tests are run only after all our containers are healthy.

For others in a similar position, I found https://github.com/nicolas-van/docker-compose-wait to be exactly what I needed.

**Comment by benesch** [comment]
It shipped! See #8777.

**Comment by ndeloof** [comment]
Yes indeed, `docker-compose up -d --wait` actually already is there :)
